### PR TITLE
Added workflow to test Next.js Server lifecycle

### PR DESCRIPTION
**Pull request by johan253**
# Added Server Lifecycle Workflow
This workflow builds a production environment and starts the Next.js server with our application. The workflow then kills the application using the Process ID (PID) generated by the `npm run start` command.

## What is the purpose?
This workflow tests that the application can start and end the Next.js server without any complications, which allows us to catch any possible bugs before deployment of the app.

### CODE REVIEW

File: .github/workflows/server.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+          node-version: '20'
+
+      - name: Install dependencies
+        run: npm install
+
+      - name: Make production build
+        run: npm run build
+
+      - name: Start Next.js server
+        run: npm run start & echo $! > server_pid.txt
```
Comment (jovanpet) [inline .github/workflows/server.yml:24]: pid_tmp.txt

File: .github/workflows/server.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+      - name: Wait for server to start
+        run: sleep 15
+
+      - name: Display server PID
+        run: cat server_pid.txt
+
```
Comment (jovanpet) [inline .github/workflows/server.yml:31]: delete pid_tmp.txr

File: .github/workflows/server.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      - name: Terminate server
+        run: kill $(cat server_pid.txt)
```
Comment (jovanpet) [inline .github/workflows/server.yml:33]: kill -9 pid

File: .github/workflows/server.yml (hunk 2 above)
Comment (jovanpet) [reply .github/workflows/server.yml:31]: This is at the end of the file.

File: .github/workflows/server.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        run: npm run build
+
+      - name: Start Next.js server
+        run: npm run start & echo $! > server_pid.txt
+
+      - name: Wait for server to start
+        run: sleep 15
+
+      - name: Display server PID
+        run: cat server_pid.txt
```
Comment (jovanpet) [inline .github/workflows/server.yml:30]: I don't think this is necessary. You can just kill the file.

**Review by veryyynice — APPROVED** [review]

**Pull request merged**